lib/Service/SynchronizationService.php

```diff
@@ -174,6 +175,30 @@ private function getOriginId(Synchronization $synchronization, array $object): i
         return $originId;
     }
 
+	public function getObjectFromSource(Synchronization $synchronization, string $endpoint): array
+	{
+		$source = $this->sourceMapper->find(id: $synchronization->getSourceId());
+
+		// Lets get the source config
+		$sourceConfig = $synchronization->getSourceConfig();
+		$headers = $sourceConfig['headers'] ?? [];
+		$query = $sourceConfig['query'] ?? [];
+		$config = [
+			'headers' => $headers,
+			'query' => $query,
+		];
+
+		if (str_starts_with($endpoint, $source->getLocation()) === true) {
+			$endpoint = str_replace(search: $source->getLocation(), replace: '', subject: $endpoint);
+		}
+
+		// Make the initial API call
+		// @TODO: method is now fixed to GET, but could end up in configuration.
```

Comment:
Just a reminder that we should avoid doing this (even though I do it myself as well 😝)
Maybe create an issue for this, or mention it in another existing issue?